pydicom/filereader.py

```diff
@@ -282,6 +294,12 @@ def data_element_generator(fp,
         if length != 0xFFFFFFFF:
             # don't defer loading of Specific Character Set value as it is
             # needed immediately to get the character encoding for other tags
+            if tag_set is not None and tag not in tag_set and tag != (
```

Comment:
Am I misunderstanding something?  If don't use a None flag, and check against empty set, then all tags will not be in the set and we will not return any.  Also, timeit above used `u==None`.  Using  `u is None` is quite a bit faster in my tests (about 0.05 usec vs 0.08 usec).